File: day1/py-sol/validate.py

dial = 50 
rotations = []
password = 0 
with open("validate.aoc") as f : 
    for line in f :
        rotations.append(line.strip())
        
for r in rotations :
    #print(r[0]) 
    if r[0] == "R":
        print(f"[Dial] {dial} Rotating right by {int(r[1:])}")
        dial += int(r[1:])
        dial = dial % 100
    else: 
        print(f"[Dial] {dial}  Rotating left by {int(r[1:])}")
        dial -= int(r[1:])
        dial = dial % 100
    print(f"[UPDATED DIAL] {dial}")
    if dial == 0 : 
            password +=1
print(dial)
print(password)
#print(rotations)

File: day1/go-sol/validate.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
)

func check(e error) {
    if e != nil {
        panic(e)
    }
}

func main() {

	file , err	:= os.Open("validate.txt")
	dial := 50
	rotations := []string{}
	password := 0
	check(err)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
        line := scanner.Text()
        rotations = append(rotations,line)
    }
	for _ , r := range rotations {
		rotation, err := strconv.Atoi(r[1:])
		check(err)

		if r[0] == 'R' {
			dial += rotation
			dial = dial % 100
		} else {
			dial -= rotation
			dial = dial % 100
		}

		if dial == 0 {
			password += 1
		}

	}

	fmt.Println(password)
	//fmt.Println(strings.Join(rotations, ", "))
    fmt.Println("Go-sol - by 0xkatana")
}
